homeassistant/components/islamic_prayer_times/config_flow.py

```diff
@@ -3,22 +3,40 @@
 
 from homeassistant import config_entries
 from homeassistant.core import callback
-
-from .const import CALC_METHODS, CONF_CALC_METHOD, DEFAULT_CALC_METHOD, DOMAIN, NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CALC_METHODS,
+    CONF_CALC_METHOD,
+    CONF_LAT_ADJ_METHOD,
+    CONF_MIDNIGHT_MODE,
+    CONF_SCHOOL,
+    CONF_TUNE,
+    DEFAULT_CALC_METHOD,
+    DEFAULT_LAT_ADJ_METHOD,
+    DEFAULT_MIDNIGHT_MODE,
+    DEFAULT_SCHOOL,
+    DOMAIN,
+    LAT_ADJ_METHODS,
+    MIDNIGHT_MODES,
+    NAME,
+    SCHOOLS,
+    TIMES_TUNE,
+)
 
 
 class IslamicPrayerFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle the Islamic Prayer config flow."""
 
-    VERSION = 1
+    VERSION = 2
```

Comment:
Why is the entry version changed? The data doesn't change, just the options right?
That can be handled without a migration. Right now, this will block people to downgrade once upgraded to this version. Thus, we should avoid bumping the version in these cases.

Reply:
So you mean I can keep the `async_migarte_entry` function to update the change in the options and keep the version as 1. The change here is very slight where I have renamed the `CALC_METHODS` values to Uppercase and I need to update this value for existing config_entries in order to pickup the correct value for this option.

Reply:
No, we don't need `async_migrate_entry`, as these are new options we can handle the default for when its missing from the entry.

Reply:
Just to clarify, `CONF_CALC_METHOD` is not a new option. This will be existing in old config entries. What changed are the possible values for the option. For example previously there was a value 'isna' which I changed to 'ISNA' because its an abbreviation. I am thinking of adding an `async_update_options` function that gets called on setup and checks and updates the existing option value.

Reply:
> For example previously there was a value 'isna' which I changed to 'ISNA' because its an abbreviation.

Those are generally handled during config entry setup.

Reply:
Just came back from a cup of coffee; thinking about it a bit. What would happen if that specific option has been converted/migrated and one downgrades... I guess it would break at that point as well.

Reply:
Actually that won't be a problem because the api coverts the value taken to lower(). So if the person downgrades and the option value was 'ISNA' the api will automatically covert it to 'isna'. The only issue I can see is that when the person tries to update the options from the UI he will not see the selected value but it will rather show as if none-selected. I guess this should be acceptable in case a person downgrades and he only needs to update the options once for it to get saved correctly.